niworkflows/interfaces/confounds.py

```diff
@@ -0,0 +1,548 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+"""
+Select terms for a confound model, and compute any requisite expansions.
+"""
+
+import re
+import numpy as np
+import pandas as pd
+from functools import reduce
+from collections import deque, OrderedDict
+from nipype.utils.filemanip import fname_presuffix
+from nipype.interfaces.base import (
+    traits, TraitedSpec, BaseInterfaceInputSpec, File, isdefined,
+    SimpleInterface
+)
+
+
+class ExpandModelInputInterface(BaseInterfaceInputSpec):
+    confounds_file = File(exists=True, mandatory=True,
+                          desc='TSV containing confound time series for '
+                          'expansion according to the specified formula.')
+    model_formula = traits.Str(
+        '(dd1(rps + wm + csf + gsr))^^2 + others', usedefault=True,
+        desc='Formula for generating model expansions. By default, the '
+        '32-parameter expansion will be generated. Note that any expressions '
+        'to be expanded *must* be in parentheses, even if they include only '
+        'a single variable (e.g., (x)^2, not x^2).')
+    output_file = traits.Str(desc='Output path')
+
+
+class ExpandModelOutputInterface(TraitedSpec):
+    confounds_file = File(exists=True, desc='Output confounds file')
+
+
+class ExpandModel(SimpleInterface):
+    """Expand a confound model according to a specified formula.
+    """
+    input_spec = ExpandModelInputInterface
+    output_spec = ExpandModelOutputInterface
+
+    def _run_interface(self, runtime):
+        if isdefined(self.inputs.output_file):
+            out_file = self.inputs.output_file
+        else:
+            out_file = fname_presuffix(
+                self.inputs.confounds_file,
+                suffix='_expansion.tsv',
+                newpath=runtime.cwd,
+                use_ext=False)
+
+        confounds_data = pd.read_csv(self.inputs.confounds_file, sep='\t')
+        _, confounds_data = parse_formula(
+            model_formula=self.inputs.model_formula,
+            parent_data=confounds_data,
+            unscramble=True
+        )
+        confounds_data.to_csv(out_file, sep='\t', index=False,
+                              na_rep='n/a')
+        self._results['confounds_file'] = out_file
+        return runtime
+
+
+class SpikeRegressorsInputInterface(BaseInterfaceInputSpec):
+    confounds_file = File(
+        exists=True, mandatory=True,
+        desc='TSV containing criterion time series (e.g., framewise '
+        'displacement, DVARS) to be used for creating spike regressors.')
+    criteria = traits.Dict(
```

Comment:
what about splitting this dict into two independent, Float inputs: `fd_thresh` and `dvars_thresh`?